src/test/java/com/znsio/sample/e2e/businessLayer/amazon/SearchBL.java

```diff
@@ -0,0 +1,59 @@
+package com.znsio.sample.e2e.businessLayer.amazon;
+
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazon.SearchResultScreen;
+import com.znsio.sample.e2e.screen.web.amazon.SearchResultScreenWeb;
+import org.apache.log4j.Logger;
+import org.assertj.core.api.SoftAssertions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SearchBL {
+
+    private static final Logger LOGGER = Logger.getLogger(SearchBL.class.getName());
+    private final TestExecutionContext context;
+    private final SoftAssertions softly;
+    private final String currentUserPersona;
+    private final Platform currentPlatform;
+
+    public SearchBL(String userPersona, Platform forPlatform){
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = userPersona;
+        this.currentPlatform = forPlatform;
+        Runner.setCurrentDriverForUser(userPersona, forPlatform, context);
+    }
+
+    public SearchBL() {
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+
+    private SearchBL verifyUserIsOnSearchResultsPage() {
+        boolean isOnSearchResultsPage = SearchResultScreenWeb.get().isUserOnSearchResultPage();
+        assertThat(isOnSearchResultsPage).as("User is still on Home page").isTrue();
+        return this;
+    }
+
+    private SearchBL verifySearchedProductListIsNotEmpty() {
+        boolean isSearchListEmpty = SearchResultScreen.get().isSearchResultEmpty();
+        assertThat(isSearchListEmpty).as("Search List is Empty for Searched Product").isFalse();
+        return this;
+    }
+
+    public ProductInformationBL userClicksOnFirstProduct() {
+        this.verifyUserIsOnSearchResultsPage().verifySearchedProductListIsNotEmpty();
+        SearchResultScreen.get().selectFirstProductFromSearchResult();
+        return new ProductInformationBL();
```

Comment:
assertion is missing

Reply:
Should I change the return type of method "SearchResultScreen.get().selectFirstProductFromSearchResult" to boolean to add assertion here or should I add an assertion for user's navigation to product detail page using

assertThat(ProductInformationScreen.get().isUserOnProductPage()).as("User is on Product Page").isTrue();

Please suggest.